src/handlers.ts

```diff
@@ -108,13 +112,36 @@ export const postFeedback = async (ctx: IRouterContext): Promise<any> => {
     throw { status: 401 }
   }
 
+  const [feedback] = await db<Feedback>('feedbacks').insert({ status, user_id: user.id }).returning('*')
+  const base64Screenshots = await Promise.all(
+    screenshots.map(async screenshot => {
+      const options = {
+        encoding: 'base64'
+      }
+      const base64Screenshot = await readFile(screenshot.path, options)
+
+      // tslint:disable-next-line: no-expression-statement
+      await db<Screenshot>('screenshots').insert({ name: screenshot.name, screenshot_file: base64Screenshot, feedback_id: feedback.id })
+      return base64Screenshot
+    })
+  )
+
   const params = {
     status,
-    screenshots,
+    screenshots: base64Screenshots,
     access_token: user.token,
     access_token_secret: user.tokenSecret
   }
 
-  const { url } = await twitter.tweetStatus(params)
+
+  // Write all screenshots to the current directory
+  // const scrshts = await db<Screenshot>('screenshots').select('*')
+  // scrshts.forEach(s => {
+  //   writeFile(`./${s.name}.png`, s.screenshot_file, { encoding: 'base64' }, () => {
+  //     console.log('saved')
+  //   })
+  // })
+
+  const { url } = { url: 'fake' } // await twitter.tweetStatus(params)
```

Comment:
As far as handling twitter and/or our database potentially being down, how does this sound for wrangling our various data stores.

Try posting to twitter
  -> If that's successful, immediately reply to the user with the tweet URL and, in the background, save the feedback in the database as a single transaction with the returned tweetUrl. If that background process fails, log the error.
  -> If that's unsuccessful, log the error and immediately reply to the user saying the tweet was not successful. The client should then retry at a later point in time. Do not save anything to the database.

If we followed this approach, the `tweetUrl` would be a non-nullable unique index for the feedback table.